### ISSUE TITLE
Add functions like onUpdate() or onChange() with old snapshots and newly added objects.

### ISSUE THREAD

**Issue by mohsinht** [tbolis/react-sketch#80]
I was using React-Sketch for an app that runs on Socket.io to transfer drawings on multiple devices at a time. Your library is a great one but it lacks such a simple function that when a user draws something on the whiteboard, some callback event must be triggered.

I tried using onChange() for this purpose but it gets a lot of calls at once, even when nothing is done on the whiteboard.

**Comment by mohsinht** [comment]
I was hoping to create something like Wyzant Online Tool. How are they doing this?

ref: https://www.wyzant.com/online/student

**Comment by rohitkrishna094** [comment]
@mohsinht Were you able to figure this out?

**Comment by mohsinht** [comment]
@rohitkrishna094 Nope. I shifted to core [FabricJS](https://github.com/fabricjs/fabric.js) for the workaround as I needed additional functionalities as well.

However I was able to add some changes in the React-Sketch library for callbacks but they had ripple effects over the other functionalities. It was becoming a headache to continue with this library. It was only good for basic stuff.
